integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java

```diff
@@ -40,4 +44,24 @@
    * @throws AuthorizationException when a problem occur with this operation.
    */
   void authorize(AuthorizationPayload authorizationPayload) throws AuthorizationException;
+
+  /**
+   * Get an access token for user to perform calls to a resource provided by a given url.
+   * @param url Resource url.
+   * @param userId User id.
+   * @return An access token.
+   * @throws AuthorizationException Thrown in any case of error.
+   */
+  String getAccessToken(String url, Long userId) throws AuthorizationException;
+
+  /**
+   * Get aprovider for user to perform calls to a resource provided by a given url.
```

Comment:
Get an OAuth1 provider for user to perform calls to an external system resource.